Wrappers/Python/cil/optimisation/functions/L1Norm.py

```diff
@@ -37,10 +36,10 @@ def soft_shrinkage(x, tau, out=None):
     --------
     the value of the soft-shrinkage operator at x: DataContainer.
     """
-    if  id(x)==id(out):
-        raise InPlaceError(message="The soft_shrinkage function cannot be used in place" )
-
     should_return = False
+    # get the sign of the input  
+    dsign = np.exp(1j*np.angle(x.as_array())) if np.iscomplexobj(x.as_array()) else x.sign()
```

Comment:
actually I just looked up the implementation and this should be safe (it implicitly uses `x.dtype.type` lol):

```suggestion
    dsign = np.exp(1j*np.angle(x.as_array())) if np.iscomplexobj(x) else x.sign()
```

Reply:
Thanks for checking @casperdcl :)